cmd/deploy/analytics.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2024 The Okteto Authors
```

Comment:
This looks like a copy of `cmd/build/v2/analytics.go` from the other PR. I think there's a lot in common and we should avoid copy/pasting this. If at some point we want to start sending other events we would need to copy/paste this yet again.

In terms of design, I think there should be a distinction between the transport and the payload.

- transport: The mechanism by which the data is sent. The k8s event, event type etc. Shared by deploy/build and others.
- payload: The actual data that is sent by the transport. This is where build and deploy will differ. They will simply have different data

The data is all serialized into json and sent in the `Note` metadata field so there's **a lot** in common between build and deploy. The payload could simply be a `json.Marshaler`. If I understand correctly, all the rest is the same.

Reply:
Agree with this. We even do things differently, for example, the k8s client provider here is passed as parameter in the `New...` function while for the build events we create it within it.

The track function has to be generic and the build and deploy operations should be the ones gathering the data and converting it to the common interface defined for the track operation